frontend/public/components/operator-hub/operator-hub-subscribe.tsx

```diff
@@ -97,18 +124,69 @@ export const OperatorHubSubscribeForm = withFormState((props: OperatorHubSubscri
     };
 
     return (!_.isEmpty(catalogSourceConfig) || hasBeenEnabled
-      ? k8sUpdate(CatalogSourceConfigModel, {...catalogSourceConfig, spec: {targetNamespace: operatorGroupNamespace, packages}}, 'openshift-marketplace', OPERATOR_HUB_CSC_NAME)
+      ? k8sUpdate(CatalogSourceConfigModel, {...catalogSourceConfig, spec: {...catalogSourceConfig.spec, packages}}, 'openshift-marketplace', OPERATOR_HUB_CSC_NAME)
       : k8sCreate(CatalogSourceConfigModel, newCatalogSourceConfig)
-    ).then(() => k8sCreate(SubscriptionModel, subscription))
+    ).then(() => props.operatorGroup.data.some(group => group.metadata.namespace === props.formState().targetNamespace)
+      ? Promise.resolve()
+      : k8sCreate(OperatorGroupModel, operatorGroup))
+      .then(() => k8sCreate(SubscriptionModel, subscription))
       .then(() => history.push('/operatorhub'));
   };
 
-  return <div>
+  const installModes = channels.find(ch => ch.name === props.formState().updateChannel).currentCSVDesc.installModes;
+  const isGlobalOperator = installModes.find(m => m.type === InstallModeType.InstallModeTypeAllNamespaces).supported;
+  const descFor = (mode: InstallModeType) => {
+    if (mode === InstallModeType.InstallModeTypeAllNamespaces && isGlobalOperator) {
+      return 'Operator will be available in all namespaces.';
+    }
+    if (mode === InstallModeType.InstallModeTypeOwnNamespace && !isGlobalOperator) {
+      return 'Operator will be available in a single namespace only.';
+    }
+    return 'This mode is not supported by this Operator';
+  };
+  const subscriptionExists = (ns: string) => props.subscription.data.some(sub => sub.metadata.namespace === ns && sub.spec.name === props.packageManifest.data.status.packageName);
+
+  return <React.Fragment>
     <div className="col-xs-6">
-      <div>
+      <React.Fragment>
         <div className="form-group">
-          <label className="co-required">Target</label>
-          <OperatorGroupSelector onChange={(target) => props.updateFormState({target})} excludeName={'olm-operators'} />
+          <label className="co-required">Installation Mode</label>
+          <div>
+            <Tooltip content={descFor(InstallModeType.InstallModeTypeAllNamespaces)} hidden={isGlobalOperator}>
```

Comment:
New section for "Create Operator Subscription" form which displays a namespace selector if the Operator package only supports single/multi namespace install mode.